IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -198,39 +226,38 @@ - (MSIDAccessToken *)getAccessTokenForAccount:(MSIDAccountIdentifier *)account
                                       context:(id<MSIDRequestContext>)context
                                         error:(NSError **)error
 {
-    NSURL *outAuthority = nil;
-    MSIDAccessToken *accessToken = (MSIDAccessToken *) [self getTokenWithType:MSIDAccessTokenType
-                                                                      account:account
-                                                                    authority:configuration.authority
-                                                                     clientId:configuration.clientId
-                                                                     familyId:nil
-                                                                       target:configuration.target
-                                                                      context:context
-                                                                        error:error
-                                                                 outAuthority:&outAuthority];
 
-    if (accessToken && !configuration.authority)
-    {
-        configuration.authority = outAuthority;
-    }
+    MSIDDefaultCredentialCacheQuery *query = [MSIDDefaultCredentialCacheQuery new];
+    query.homeAccountId = account.homeAccountId;
+    query.environment = configuration.authority.msidHostWithPortIfNecessary;
+    query.realm = configuration.authority.msidTenant;
+    query.clientId = configuration.clientId;
+    query.target = configuration.target;
+    query.targetMatchingOptions = SubSet;
```

Comment:
Should we add MSID prefix to the ENUM as we did for other ENUMs in the project? :)